### PR TITLE
Sahar/branch op support

### PR DESCRIPTION
**Pull request by sfatimar**
pulling in changes for argmax and argmin ops which only support fp32 and bf16 for first input

-It solves the test failures where int32 was the first input and was unsupported by openvino 
-also adds supported ops for gpu and vpu and sets the io precision for gpu and vpu

### CODE REVIEW

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc [hunk 1]
```diff
@@ -127,6 +125,7 @@ bool IsOpSupported(std::string name, std::string device) {
       "Relu",
       "Reshape",
       "Shape",
+      "Selu",
```
Comment (suryasidd) [inline onnxruntime/core/providers/openvino/openvino_execution_provider.cc:128]: Can you please add ops in alphabetical order, these ops are arranged alphabetically for easier readability.

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc [hunk 2]
```diff
@@ -136,6 +135,7 @@ bool IsOpSupported(std::string name, std::string device) {
       "Sum",
       "Tanh",
       "TopK",
+      "Erf",
```
Comment (suryasidd) [inline onnxruntime/core/providers/openvino/openvino_execution_provider.cc:138]: Same here

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc [hunk 3]
```diff
@@ -165,8 +165,19 @@ bool IsOpSupported(std::string name, std::string device) {
 
   std::set<std::string> supported_ops_gpu = {
     "HardSigmoid",
+    "Abs",
+    "Asin",
+    "Asinh",
+    "Atan",
+    "HardSigmoid",
```
Comment (suryasidd) [inline onnxruntime/core/providers/openvino/openvino_execution_provider.cc:172]: HardSigmoid is added twice

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc [hunk 4]
```diff
@@ -411,7 +422,14 @@ static bool IsUnsupportedOpMode(const Node* node, const onnxruntime::GraphViewer
     // nGraph only supports constant shape input values
     const auto& shape_input = node->InputDefs()[1];
     return !graph_viewer.IsConstantInitializer(shape_input->Name(), true);
-  }
+  } else if (optype == "ArgMax" || optype == "ArgMin" ) {
+      
+    auto dtype = node->InputDefs()[0]->TypeAsProto()->tensor_type().elem_type();
+    if (!(dtype == ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_FLOAT ||
+          dtype == ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_BFLOAT16)) {
```
Comment (suryasidd) [inline onnxruntime/core/providers/openvino/openvino_execution_provider.cc:429]: We anyway don't support BFLOAT16. So just have it check for float

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc (hunk 1 above)
Comment (sfatimar) [reply onnxruntime/core/providers/openvino/openvino_execution_provider.cc:128]: Done

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc (hunk 2 above)
Comment (sfatimar) [reply onnxruntime/core/providers/openvino/openvino_execution_provider.cc]: done

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc (hunk 3 above)
Comment (sfatimar) [reply onnxruntime/core/providers/openvino/openvino_execution_provider.cc:171]: done

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc (hunk 4 above)
Comment (sfatimar) [reply onnxruntime/core/providers/openvino/openvino_execution_provider.cc]: done

**Pull request merged**